### PR TITLE
New: Allow modifying base config (fixes #2143)

### LINKED ISSUE #2143: Allow to modify base configuration
I'm implementing a library which help developers to quickly set up a project. ESLint configuration will be pre-defined by the library and can be overrided by the developer also.

For example, the library want to set following configuration by default:

```js
{
  "parser": "babel-eslint",
  "rules": [
    "quotes": [ 2, "single" ],
    "strict": [ 2, "never" ]
  ]
}
```

And the developer has the option to override it (through `.eslintrc`, `package.json`, etc.):

```js
{
  "parser": "espree",
  "rules": [
    "strict": [ 2, "global" ]
  ]
}
```

The above example is taken from a real use case, in which the whole project is supposed to be written in ES6/7, while some directories have files which are written in ES5 and down.

I digged through ESLint code and realize that at the moment there is no way to solve that use case. I propose a solution as:

* Adding a new option called `baseConfig`.
* After initialize `configHelper.baseConfig`, merge it with `baseConfig` option.

Please refer to this pull request #2141 for more detail. Thank you.

### PR DESCRIPTION
**Pull request by meoguru**

### CODE REVIEW

File: docs/user-guide/command-line-interface.md [hunk 1]
```diff
@@ -28,6 +28,7 @@ Options:
   -f, --format String         Use a specific output format - default: stylish
   -v, --version               Outputs the version number
   --reset                     Set all default rules to off - default: false
+  --base-config Boolean | path::String  Enable/disable base config, could be passed a file path - default: true
```
Comment (nzakas) [inline docs/user-guide/command-line-interface.md]: I'd prefer not to expose this on the command line and only expose it on `CLIEngine`.

File: lib/cli-engine.js [hunk 2]
```diff
@@ -69,6 +71,7 @@ var fs = require("fs"),
 var defaultOptions = {
         configFile: null,
         reset: false,
+        baseConfig: true,
```
Comment (nzakas) [inline lib/cli-engine.js]: default value should be what's in `eslint.json`.

File: lib/config.js [hunk 3]
```diff
@@ -215,15 +215,29 @@ function createEnvironmentConfig(envs, reset) {
...
+
     options = options || {};
 
     this.ignore = options.ignore;
     this.ignorePath = options.ignorePath;
     this.cache = {};
 
-    this.baseConfig = options.reset ? { rules: {} } :
-            require(path.resolve(__dirname, "..", "conf", "eslint.json"));
+    useBaseConfig = options.baseConfig;
```
Comment (nzakas) [inline lib/config.js]: Please add some comments here explaining this behavior - it's not very obvious just from looking at it.

**Comment by meoguru** [comment]
Close and move to #2197.

**Pull request closed without merging**